src/test/util_tests.cpp

```diff
@@ -1235,9 +1238,9 @@ BOOST_AUTO_TEST_CASE(util_ParseMoney)
     BOOST_CHECK(!ParseMoney("-1", ret));
 
     // Parsing strings with embedded NUL characters should fail
-    BOOST_CHECK(!ParseMoney(std::string("\0-1", 3), ret));
-    BOOST_CHECK(!ParseMoney(std::string("\01", 2), ret));
-    BOOST_CHECK(!ParseMoney(std::string("1\0", 2), ret));
+    BOOST_CHECK(!ParseMoney("\0-1"s, ret));
+    BOOST_CHECK(!ParseMoney("\0" "1"s, ret));
```

Comment:
This change strikes me as odd, shouldn't it be `"\01"s` or did you intentionally want the other approach here to improve the test?

Reply:
I guess the intention of `std::string("\01", 2)` was to test the string `{0, 49}` (ASCII code of `'1'` is `49`). But this is not what it did because `"\01"` is the same as `"\1"` and so the resulting string is `{1, 0}` (`1` from `\1` and the terminating nul).